src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/ContigPloidyAnnotationTableReader.java

```diff
@@ -0,0 +1,131 @@
+ * Reads contig ploidy annotations from tab-separated files and readers.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class ContigPloidyAnnotationTableReader extends TableReader<ContigPloidyAnnotation> {
+
+    public static final Logger logger = LogManager.getLogger(ContigPloidyAnnotationTableReader.class);
+
+    /**
+     * The set of ploidy tags (= sex genotype identifer strings)
+     */
+    private final Set<String> ploidyTagsSet;
+
+    /**
+     * Public constructor.
+     *
+     * @param sourceName name of the source
+     * @param sourceReader an instance of {@link Reader}
+     * @throws IOException if a reading error occurs
+     */
+    public ContigPloidyAnnotationTableReader(final String sourceName, @Nonnull final Reader sourceReader)
+            throws IOException {
+        super(sourceName, sourceReader);
+        TableUtils.checkMandatoryColumns(columns(), ContigPloidyAnnotationTableColumn.MANDATORY_CONTIG_ANNOTATION_COLUMNS,
+                UserException.BadInput::new);
+        ploidyTagsSet = Sets.difference(new HashSet<>(columns().names()),
+                ContigPloidyAnnotationTableColumn.MANDATORY_CONTIG_ANNOTATION_COLUMNS_SET);
+        if (ploidyTagsSet.size() == 0) {
+            throw new UserException.BadInput("At least one ploidy column is required!");
+        }
+        logger.info("Ploidy tags: " + ploidyTagsSet.stream().collect(Collectors.joining(", ")));
+    }
+
+    /**
+     * Public constructor.
+     *
+     * @param sourceReader an instance of {@link Reader}
+     * @throws IOException if a reading error occurs
+     */
+    public ContigPloidyAnnotationTableReader(@Nonnull final Reader sourceReader) throws IOException {
+        this(null, sourceReader);
+    }
+
+    /**
+     * Creates a {@link ContigPloidyAnnotation} instance from a {@link DataLine}
+     *
+     * @param dataLine a data line
+     * @return an instance of {@link ContigPloidyAnnotation}
+     */
+    @Override
+    protected ContigPloidyAnnotation createRecord(@Nonnull final DataLine dataLine) {
+        final String contigName = dataLine.get(ContigPloidyAnnotationTableColumn.CONTIG_NAME);
+
+        final ContigClass contigClass;
+        final String contigClassString = dataLine.get(ContigPloidyAnnotationTableColumn.CONTIG_CLASS);
+        if (!ContigClass.CONTIG_CLASS_NAMES_SET.contains(contigClassString)) {
+            throw new UserException.BadInput("Bad contig class: provided value: " + contigClassString + ", acceptable values: " +
+                    ContigClass.CONTIG_CLASS_NAMES_SET.stream().collect(Collectors.joining(", ", "[", "]")));
+        } else {
+            contigClass = ContigClass.valueOf(contigClassString);
+        }
+
+        final Map<String, Integer> ploidyMap = new HashMap<>();
+        try {
+            /* all lines must all ploidy annotations defined in the header */
+            ploidyTagsSet.forEach(tag -> ploidyMap.put(tag, dataLine.getInt(tag)));
```

Comment:
Not necessary. All subclasses of `Collection<T>` implement `Iterable<T>` which provides `void forEach(Consumer<? super T>` for them.